tests/integration/test_recipe.py

```diff
@@ -194,6 +194,40 @@ def get_recipe(tempdir, content, cfg):
     return recipe
 
 
+def test_recipe_no_datasets(tmp_path, config_user):
+    content = dedent("""
+        preprocessors:
+          preprocessor_name:
+            extract_levels:
+              levels: 85000
+              scheme: nearest
+
+        diagnostics:
+          diagnostic_name:
+            variables:
+              ta:
+                preprocessor: preprocessor_name
+                project: CMIP5
+                mip: Amon
+                exp: historical
+                ensemble: r1i1p1
+                start_year: 1999
+                end_year: 2002
+            scripts: null
+        """)
+    exc_message = \
+        "You have not specified any dataset " + \
+        "or additional_dataset groups for variable " + \
+        "{'preprocessor': 'preprocessor_name', 'project': 'CMIP5'," + \
+        " 'mip': 'Amon', 'exp': 'historical', 'ensemble': 'r1i1p1'" + \
+        ", 'start_year': 1999, 'end_year': 2002, 'variable_group':" + \
+        " 'ta', 'short_name': 'ta', 'diagnostic': " + \
+        "'diagnostic_name'} Exiting."
```

Comment:
```suggestion
    exc_message = (
        "You have not specified any dataset "
        "or additional_dataset groups for variable "
        "{'preprocessor': 'preprocessor_name', 'project': 'CMIP5',"
        " 'mip': 'Amon', 'exp': 'historical', 'ensemble': 'r1i1p1'"
        ", 'start_year': 1999, 'end_year': 2002, 'variable_group':"
        " 'ta', 'short_name': 'ta', 'diagnostic': "
        "'diagnostic_name'} Exiting.")
```
Sorry to keep bugging you with string formatting, but can you please avoid using all these backslashes and just use brackets instead?

Reply:
nay bother man, good point! :beer: